### PR TITLE
Using lists to define instructions about how to output predictions

### PR DESCRIPTION
**Pull request by Joao-L-S-Almeida**
* Using a string, as `"prediction"`  to output just predictions.
* Using a list as `["prediction", "probabilities"]`  to output both predictions and `probabilities/logits`.

### CODE REVIEW

**Review by romeokienzler — DISMISSED** [review]

File: terratorch/tasks/segmentation_tasks.py [hunk 1]
```diff
@@ -151,21 +151,43 @@ class or the probabilities for all of them. This argument has been deprecated an
...
+                              }
+
+        # `output_on_inference` can be a list or a string.
+        if isinstance(output_on_inference, list):
+            list_of_selectors = ()
+            for var in output_on_inference:
+                if var in self.operation_map:
+                    list_of_selectors += (self.operation_map[var],)
+                else:
+                    warnings.warn(f"Option {var} is not supported.")
```
[the remark this answers is not in the document]
Comment (Joao-L-S-Almeida) [reply terratorch/tasks/segmentation_tasks.py:179]: In fact. I was considering that the execution could continue for the others variables, but maybe it's better to interrupt and force the user to fix it.

File: tests/resources/configs/manufactured-finetune_prithvi_swin_B_segmentation.yaml [hunk 2]
```diff
@@ -99,8 +99,10 @@ model:
       num_frames: 1
       num_classes: 2
       head_dropout: 0.5708022831486758
-        #output_on_inference: probabilities #both
-    output_most_probable: False
+    output_on_inference:
```
Comment (romeokienzler) [inline tests/resources/configs/manufactured-finetune_prithvi_swin_B_segmentation.yaml:102]: @Joao-L-S-Almeida we need - if we add functionality to the config file - always update

    - documentation
    - json schema

File: terratorch/tasks/segmentation_tasks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Joao-L-S-Almeida) [reply terratorch/tasks/segmentation_tasks.py]: Updated it in the last commit.

**Review by romeokienzler — APPROVED** [review]
